src/main/java/net/neoforged/neoforge/event/entity/living/ProjectileWeaponShootEvent.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.entity.living;
+
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.projectile.Projectile;
+import net.minecraft.world.item.BowItem;
+import net.minecraft.world.item.CrossbowItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.ProjectileWeaponItem;
+import net.minecraft.world.item.component.ChargedProjectiles;
+import net.minecraft.world.level.Level;
+import net.neoforged.bus.api.ICancellableEvent;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * ProjectileWeaponShootEvent is fired when a living entity shoots a {@link Projectile} with a {@link ProjectileWeaponItem}.
+ */
+public abstract class ProjectileWeaponShootEvent extends LivingEvent {
+    private final @Nullable LivingEntity target;
+    private final ItemStack weaponItem;
+    private final ItemStack projectileItem;
+
+    public ProjectileWeaponShootEvent(LivingEntity entity, @Nullable LivingEntity target, ItemStack weaponItem, ItemStack projectileItem) {
```

Comment:
mark this as `@ApiStatus.Internal`

Reply:
Made it package private, basically the same effect as `@ApiStatus.Internal`.